support/oidc-discovery-provider/handler.go

```diff
@@ -34,14 +44,31 @@ func (h *Handler) serveWellKnown(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if !h.domains[r.Host] {
```

Comment:
Only serves content when requested via a preconfigured domain, this eleminates rougue proxies or other tricks in the infrastructure.